nodes/clarify_util.js

```diff
@@ -130,22 +136,30 @@ function validateString(validationErrors, varName, variable) {
     validationErrors.push(varName + ' must be string');
     return;
   }
+}
 
-  // Field spesific validations
-  let allowed = [];
-  switch (varName) {
-    case 'type':
-      allowed = ['enum', 'numeric'];
-      if (!allowed.includes(variable)) {
-        validationErrors.push('unsupported type: ' + variable);
-      }
-      break;
-    case 'sourceType':
-      allowed = ['aggregation', 'measurement', 'prediction'];
-      if (!allowed.includes(variable)) {
-        validationErrors.push('unsupported sourceType: ' + variable);
-      }
-      break;
+function validateStringEnum(validationErrors, varName, variable, allowed) {
+  validationErrors = validateString(validationErrors, varName, variable);
+  if (validationErrors.length > 0) {
+    return;
+  }
+
+  if (!allowed.includes(variable)) {
+    validationErrors.push(`unsupported ${varName}: ${variable}`);
+    return;
+  }
+}
+
+function validateStringRFC3339(validationErrors, varName, variable) {
+  validationErrors = validateString(validationErrors, varName, variable);
+  if (validationErrors.length > 0) {
+    return;
+  }
+
+  let d = Duration.fromISO(variable);
```

Comment:
We differ between "fixed" (not including months or years), and "variadic" (including months or years) duration in the API. For the current version, we only support the _fixed duration_ variant as that is what the design was made to display in the admin UI. This method might not validate that.

Here is the regex we use:

```go
const patternWeekToFraction = `^P((?P<weeks>\d+)W)?((?P<days>\d+)D)?(T((?P<hours>\d+)H)?((?P<minutes>\d+)M)?((?P<fractions>\d+(\.\d+)?)S)?)?$`
```

Reply:
- Weeks are interpreted as 7 days.
- Days are interpreted as 24 hours.
- Months and years cause an error.

Reply:
Can we maybe add a different check (after the isValid) to simply see if month and/or year is set, and than give the an error explaining we only accept a _fixed duration_?

PS! It could be relevant to allow `sampleInterval` to hold a variable length duration, but we where not able to do that for self-service launch as it would require some level of UI support.

Reply:
I have updated the code to check if year or month is set.